airbyte-integrations/connectors/destination-azure-blob-storage/src/main/java/io/airbyte/integrations/destination/azure_blob_storage/file/uploader/AbstractS3AzureUploader.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.azure_blob_storage.file.uploader;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.DeleteObjectsRequest;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.specialized.AppendBlobClient;
+import io.airbyte.integrations.destination.s3.S3DestinationConfig;
+import io.airbyte.integrations.destination.s3.writer.S3Writer;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.DestinationSyncMode;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractS3AzureUploader<T extends S3Writer> extends AbstractAzureUploader<S3Writer> {
```

Comment:
why do we need to have both AbstractS3AzureUploader and AbstractGcsAzureUploader? they look like duplicates with a small difference in the AmazonS3 client initialization

Reply:
updated it